plugin/pkg/auth/authorizer/node/graph.go

```diff
@@ -94,6 +95,8 @@ type namespaceVertexMapping map[string]nameVertexMapping
 type nameVertexMapping map[string]*namedVertex
 
 func NewGraph() *Graph {
+	RegisterMetrics()
```

Comment:
we should not register global metrics as a side-effect of creating what was previously an isolated object. metrics should be set up in a separate path we are sure is a singleton in the process

Reply:
an alternative place to instrument this would be in the graph populator (which could take an already-registered `*HistogramVec` parameter), where we wire up informers to call the Add/Update/Delete* methods

Reply:
Why can't we simply register it in init() function in this package? There are already multiple places in our codebase doing that, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/metrics.go#L79

Reply:
I'm -1 on adding more places where we register in init()... I consider that an anti-pattern we should get rid of, not extend

Reply:
I copied this pattern from endpointslice controller: https://github.com/kubernetes/kubernetes/blob/eb9bca5f060ab1410a5228dc77026986e32505eb/pkg/controller/endpointslice/endpointslice_controller.go#L93

but I agree, it's quite ugly. I will check what other components do.

Reply:
@liggitt - for my own education, why do you consider that anti-pattern? init() guarantees you "exactly once" and guarantees metrics registration if a given package is used.

Reply:
because it registers metrics for anyone that imports the package, which isn't always desired

Reply:
Other places often use func init() { legacyregistry.MustRegister() }

I migrated to this pattern.

Reply:
I added a comment below for an alternative (preferred) pattern.

Reply:
I used alternative approach suggested below.